app/services/settings/update_service.rb

```diff
@@ -47,6 +47,10 @@ def set_setting_value(name, value)
     Setting[name] = new_value
     if name == :work_package_done_ratio && old_value != "status" && new_value == "status"
       WorkPackages::Progress::ApplyStatusesChangeJob.perform_later(cause_type: "progress_mode_changed_to_status_based")
+    elsif name == :total_percent_complete_mode && old_value != "work_weighted_average" && new_value == "work_weighted_average"
+      WorkPackages::Progress::ApplyTotalPercentCompleteModeChangeJob
+        .perform_later(mode: new_value,
+                       cause_type: "total_percent_complete_mode_changed_to_work_weighted_average")
```

Comment:
Ah, I did not think enough for this one. It's called once for each modified setting, so it will be called for both and 2 jobs will be triggered.
It's probably not ideal to have the journal being updated twice, but that will do for now.